app/repositories/user_repository.rb

```diff
@@ -0,0 +1,56 @@
+class UserRepository < Hanami::Repository
+  def sign_up(session: nil, **create_params)
```

Comment:
Do `sign_up` and `authenticate` really belong in the Repository? I thought it was meant just for data retrieval, not actions like those. I would expect something along the lines of `User::SignUp.call` and `User::Authenticate.call`. Or maybe `Authentication.new.sign_up`.

Reply:
So, I *have* a `User::Create` and `Session::Create`. But I wanted all the authentication (password hashing and checking) done in one place. I could see if I could extract it to another dedicated class//module that the others rely on.

Reply:
I think some additional layer of abstraction is warranted.

Reply:
I'll take a look at it this weekend.